lua_modules/fifo/fifosock.lua

```diff
@@ -58,7 +58,15 @@ local function wrap(sock)
     -- Either that was a function or we've hit our coalescing limit or
     -- we didn't ship above.  Ship now, if there's something to ship.
     if s ~= nil then
-      if sslan == 0 then sock:send(s) else sock:send(ssla .. s) end
+      if sslan == 0 then
+        if #s > 0 then
```

Comment:
Sure, avoiding `send`-ing 0-length things seems like a good idea.  And while `return`-ing `ns or nil, true` below means we won't call `gc()` this time, we will probably do so "soon" since we're about to be called back by the `fifo` logic.